### PR TITLE
Prevent users from sending mails using RoDEX on maps with nostorage mapflag.

### PR DESCRIPTION
**Pull request by pazkero**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Prevent users from sending mails using RoDEX on maps with nostorage mapflag.

Makes sense on my head but **I have not checked official behavior**, thus I advise checking if it is possible to use RoDEX not only to send but also to receive mails on these maps.

I would kindly appreciate if the PR was closed if this is not a desired or official behavior.

**Issues addressed:** None

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by 4144** [comment]
why connecting rodex with no storage flag?
if need restrict rodex, may be better add ``nomail`` map flag?

**Comment by pazkero** [comment]
..I guess you are right.-- Sent from Yandex.Mail for mobile17:51, March 9, 2021, Andrei Karas <notifications@github.com>:
why connecting rodex with no storage flag?
if need restrict rodex, may be better add nomail map flag?

**Review by Gardosen — APPROVED** [review]
approved.

**Comment by dastgirp** [comment]
> why connecting rodex with no storage flag?
> if need restrict rodex, may be better add `nomail` map flag?

Maybe adding a battle_config is good in this case?

Not so well known in most servers, players use rodex as alternate storage, and retrieves item from rodex wherever storage is not allowed.

Rather than sending mails, it would be better to restrict retrieval of items for maps with nostorage.

Or always sync nostorage with new mapflag that doesn't allow rodex

File: src/map/rodex.c [hunk 1]
```diff
@@ -236,6 +236,11 @@ static int rodex_send_mail(struct map_session_data *sd, const char *receiver_nam
 		return RODEX_SEND_MAIL_FATAL_ERROR;
 	}
 
+	if (!pc_has_permission(sd, PC_PERM_BYPASS_NOSTORAGE) && (map->list[sd->bl.m].flag.nostorage & 2)) {
+		rodex->clean(sd, 1);
+		return RODEX_SEND_MAIL_FATAL_ERROR;
+    }
```
Comment (EPuncker) [inline src/map/rodex.c:242]:
```suggestion
	}
```

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**